test/rest_api/test_storage_service.py

```diff
@@ -349,3 +349,19 @@ def test_storage_service_snapshot_mv_si(cql, this_dc, rest_api):
                         pytest.fail(f"Snapshot of secondary index {si} should have failed")
                 except requests.HTTPError:
                     pass
+
+# ...but not when the snapshot is automatic (pre-scrub).
+def test_materialized_view_pre_scrub_snapshot(cql, this_dc, rest_api):
+    with new_test_keyspace(cql, f"WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', '{this_dc}' : 1 }}") as keyspace:
+        schema = 'p int, v text, primary key (p)'
+        with new_test_table(cql, keyspace, schema) as table:
+            stmt = cql.prepare(f"INSERT INTO {table} (p, v) VALUES (?, ?)")
+            cql.execute(stmt, [0, 'hello'])
+
+            with new_materialized_view(cql, table, '*', 'v, p', 'v is not null and p is not null') as mv:
+                resp = rest_api.send("GET", f"storage_service/keyspace_scrub/{keyspace}")
```

Comment:
Is there a way we verify that snaoshot was actually taken on the view?
Does the test fail here without the fix?

Reply:
- we can tag the snapshots and then parse the output of "POST /storage_service/snapshots" somehow, presumably.  but I'm inclined to trust the absence of error here :)
- yes (HTTP 500)

Reply:
The absence of error reproduces the previous bug (which was an error), so in that sense it is a good regression test - but it does not verify that the snapshot was taken (the code could have simply silently not snapshotted the view). I'll leave it up to you to decide which is better to test. I'm not even sure that snapshotting the view is a good idea - I'll mention this in another comment.

Reply:
I think it's good enough since the original issue (#10760) is about scrub failing and it might be a legitimate behavior to just skip the snapshot.  IMO taking the snapshot of the view before scrub is potentially beneficial for recovering from theoretical disasters - i.e. to be on the safe side.

Note that #9884 will also include snapshot of views, in addition to snapshots of deleted tables, for snapshot-ctl purposes so they are listed and can be deleted.